cypress/e2e/calendar/user-permissions-view-calendars.cy.js

```diff
@@ -0,0 +1,69 @@
+import {
+  MultiColumnListCell, PaneHeader, Pane, Button,
+  including, Link
+} from '../../../interactors';
+
+
+import calendarFixtures from '../../fixtures/calendar_e2e_fixtures';
+import permissions from '../../support/dictionary/permissions';
+
+const testServicePoint = calendarFixtures.servicePoint;
+const testCalendar = calendarFixtures.calendar;
+
+
+
+describe('User with Settings (Calendar): Can view existing calendars', () => {
+  let testCalendarResponse;
+  before(() => {
+    // login as admin so necessary state can be created
+    cy.loginAsAdmin();
+    // reset db state
+    cy.deleteServicePoint(testServicePoint.id, false);
+
+    // create test service point
+    cy.createServicePoint(testServicePoint, (response) => {
+      testCalendar.assignments = [response.body.id];
+
+      cy.createCalendar(testCalendar, (calResponse) => {
+        testCalendarResponse = calResponse.body;
+      });
+    });
+    cy.logout();
+
+    cy.openCalendarSettings();
+    cy.createTempUser([
+      permissions.calendarView.gui,
+    ]).then(userProperties => {
+      cy.login(userProperties.username, userProperties.password);
+      cy.openCalendarSettings();
+    });
+  });
+
+  after(() => {
+    cy.logout();
+
+    // // login as admin to teardown testing data
+    cy.loginAsAdmin();
+    cy.deleteServicePoint(testServicePoint.id, true);
+    cy.deleteCalendar(testCalendarResponse.id);
+  });
+
+
+  it('views existing calendar', () => {
+    const PaneActionButton = Button({ className: including('actionMenuToggle') });
+    cy.do([
```

Comment:
don't use cy.do inside test. Use it inside fragments please.
Create a common funtion in fragments, import it in test and use it